rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -193,9 +193,6 @@ rmw_create_publisher(
 
   return publisher;
 fail:
-  if (publisher) {
```

Comment:
Wouldn't this be the case when the goto from the following line triggers: https://github.com/ros2/rmw_opensplice/pull/207/files#diff-2184ec33b792d378e335481d04c19f40R190